### ISSUE TITLE
Crafting Byproduct Extraction Upgrade Improvements

### ISSUE THREAD

**Issue by mikeloeven** [RS485/LogisticsPipes#847]
Three improvements should be made to the crafting byproduct extraction upgraded.

Support Stacking of these modules 

Each additional upgrade installed should add one item slot for procedures that have multiply byproducts

Toggle Button to extract byproducts during crafting operation 

Currently the card only operates after a successful crafting request however for large requests the byproducts can backstuff the machine and cause it to shut down hanging the entire system. when crafting with machines byproducts need to be extracted as they are produced not at the end of a crafting operation 

Lastly Chassis Support 
 cannot use this upgrade with crafting modules that are installed in a chassis as the chassis will not accept this upgrade

**Comment by UntouchedWagons** [comment]
Yeah I think I need this module to automate Forestry's Thermionic Fabricator properly. The TF makes four tubes at a time but if only two are needed, two will be pulled out of the result slot and the other two will be put into the TF's inventory effectively wasting some of the result items.

**Comment by bookerthegeek** [comment]
If your using a chassis then place an advanced extractor and set it to pull what you are crafting out. This will pull all the extra crafted items out of the table. Or blacklist the the crafting mats from being extracted would also work.

**Comment by UntouchedWagons** [comment]
I'll give that a try.

**Comment by UntouchedWagons** [comment]
Didn't work. I replaced the MK2 chassis with an MK3 chassis, added an Advanced Extractor Module to pull Lapis Electron Tubes and requested 2, 2 tubes were left in the machine's inventory.

**Comment by bookerthegeek** [comment]
Did you set the extractor to the correct sneaky side? I seem to recall forestry machines being side specific for extracting. I'll post a screenshot for you later when I get home from work.

**Comment by bookerthegeek** [comment]
And when you say inventory, any we talking the output inventory or the input inventory?

**Comment by UntouchedWagons** [comment]
I'm referring to the input inventory. I'll try the sneaky option.

**Comment by UntouchedWagons** [comment]
The sneaky option did not work. It's odd because a BC Gold AND Gate can detect if there's lapis and redstone in the same inventory, but for some reason nothing can pull out of that same inventory.
